packages/react/src/components/layout/CarouselView.tsx

```diff
@@ -0,0 +1,54 @@
+import { isTrackBundlePinned, TrackBundleFilter } from '@livekit/components-core';
+import { Track } from 'livekit-client';
+import * as React from 'react';
+import { useMaybeLayoutContext } from '../../context';
+import { useSize } from '../../helper';
+import { useTracks, useVisualStableUpdate } from '../../hooks';
+import { TrackLoop } from '../TrackLoop';
+
+const MIN_HEIGHT = 120;
+const MIN_WIDTH = 150;
+export interface CarouselViewProps extends React.HTMLAttributes<HTMLMediaElement> {
+  filter?: TrackBundleFilter;
+  filterDependencies?: [];
+}
+
+export function CarouselView({ filter, filterDependencies = [], ...props }: CarouselViewProps) {
+  const asideEl = React.useRef<HTMLDivElement>(null);
+  const layoutContext = useMaybeLayoutContext();
+  const trackBundles = useTracks([
+    { source: Track.Source.Camera, withPlaceholder: true },
+    { source: Track.Source.ScreenShare, withPlaceholder: false },
+  ]);
+  const filteredTiles = React.useMemo(() => {
+    const tilesWithoutPinned = trackBundles.filter(
+      (tile) => !layoutContext?.pin.state || !isTrackBundlePinned(tile, layoutContext.pin.state),
+    );
+    return filter ? tilesWithoutPinned.filter(filter) : tilesWithoutPinned;
+  }, [filter, layoutContext?.pin.state, trackBundles, ...filterDependencies]);
+
+  const { width, height } = useSize(asideEl);
+  const isVertical = height >= width;
```

Comment:
I think this is good as a guess, but maybe better if we allow to (additionally) override this via a prop setting?

Reply:
Good point, but since we would also have to change some things in the css, I'll create a new issue for that.